src/lib.rs

```diff
@@ -852,20 +853,31 @@ mod simd {
 
 /// A type with no alignment requirement.
 ///
-/// A `Unalign` wraps a `T`, removing any alignment requirement. `Unalign<T>`
+/// An `Unalign` wraps a `T`, removing any alignment requirement. `Unalign<T>`
 /// has the same size and ABI as `T`, but not necessarily the same alignment.
 /// This is useful if a type with an alignment requirement needs to be read from
 /// a chunk of memory which provides no alignment guarantees.
 ///
 /// Since `Unalign` has no alignment requirement, the inner `T` may not be
-/// properly aligned in memory, and so `Unalign` provides no way of getting a
-/// reference to the inner `T`. Instead, the `T` may only be obtained by value
-/// (see [`get`] and [`into_inner`]).
+/// properly aligned in memory. There are four ways to access the inner `T`:
+/// - by value, using [`get`] or [`into_inner`]
+/// - fallibly by reference, using [`try_deref`] or [`try_deref_mut`]; these can
+///   fail if the `Unalign` does not satisfy `T`'s alignment requirement at
+///   runtime
+/// - unsafely by reference, using [`deref_unchecked`] or
+///   [`deref_mut_unchecked`]; it is the caller's responsibility to ensure that
+///   the `Unalign` satisfies `T`'s alignment requirement
+/// - (where `T: Unaligned`) infallibly by reference, using [`Deref::deref`] or
+///   [`DerefMut::deref_mut`]
 ///
 /// [`get`]: Unalign::get
 /// [`into_inner`]: Unalign::into_inner
+/// [`try_deref`]: Unalign::try_deref
+/// [`try_deref_mut`]: Unalign::try_deref_mut
+/// [`deref_unchecked`]: Unalign::deref_unchecked
+/// [`deref_mut_unchecked`]: Unalign::deref_mut_unchecked
 #[allow(missing_debug_implementations)]
-#[derive(FromBytes, Unaligned, Copy)]
+#[derive(FromBytes, Unaligned, Default, Copy)]
 #[repr(C, packed)]
 pub struct Unalign<T>(T);
```

Comment:
Not actionable, just a note: we don't need to write a custom drop that aligns the inner value before dropping. The compiler generates a shim that does that for us. See https://github.com/rust-lang/reference/issues/488

Reply:
Ack 👍 

Were you thinking it'd be worth putting that in a comment or something for future readers? I wouldn't be opposed.

Reply:
Not sure, it could be worth documenting so the knowledge isn't lost though. I struggled to find any documentation on how packed drops work.

Reply:
Yeah, sounds reasonable. Added.